builtin/providers/aws/resource_aws_alb.go

```diff
@@ -0,0 +1,348 @@
+	elbconn := meta.(*AWSClient).elbv2conn
+
+	elbOpts := &elbv2.CreateLoadBalancerInput{
+		Name: aws.String(d.Get("name").(string)),
+		Tags: tagsFromMapELBv2(d.Get("tags").(map[string]interface{})),
+	}
+
+	if scheme, ok := d.GetOk("internal"); ok && scheme.(bool) {
+		elbOpts.Scheme = aws.String("internal")
+	}
+
+	if v, ok := d.GetOk("security_groups"); ok {
+		elbOpts.SecurityGroups = expandStringList(v.(*schema.Set).List())
+	}
+
+	if v, ok := d.GetOk("subnets"); ok {
+		elbOpts.Subnets = expandStringList(v.(*schema.Set).List())
+	}
+
+	log.Printf("[DEBUG] ALB create configuration: %#v", elbOpts)
+	var albArn string
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		resp, err := elbconn.CreateLoadBalancer(elbOpts)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		if len(resp.LoadBalancers) != 1 {
+			return resource.NonRetryableError(fmt.Errorf(
+				"No loadbalancers returned following creation of %s", d.Get("name").(string)))
+		}
+
+		albArn = *resp.LoadBalancers[0].LoadBalancerArn
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(albArn)
+	log.Printf("[INFO] ALB ID: %s", d.Id())
+
+	return resourceAwsAlbUpdate(d, meta)
+}
+
+func resourceAwsAlbRead(d *schema.ResourceData, meta interface{}) error {
+	elbconn := meta.(*AWSClient).elbv2conn
+	albArn := d.Id()
+
+	describeAlbOpts := &elbv2.DescribeLoadBalancersInput{
+		LoadBalancerArns: []*string{aws.String(albArn)},
+	}
+
+	describeResp, err := elbconn.DescribeLoadBalancers(describeAlbOpts)
+	if err != nil {
+		if isLoadBalancerNotFound(err) {
+			// The ALB is gone now, so just remove it from the state
+			d.SetId("")
+			return nil
+		}
+
+		return errwrap.Wrapf("Error retrieving ALB: {{err}}", err)
+	}
+	if len(describeResp.LoadBalancers) != 1 {
+		return fmt.Errorf("Unable to find ALB: %#v", describeResp.LoadBalancers)
+	}
+
+	alb := describeResp.LoadBalancers[0]
+
+	d.Set("name", alb.LoadBalancerName)
+	d.Set("internal", (alb.Scheme != nil && *alb.Scheme == "internal"))
+	d.Set("security_groups", flattenStringList(alb.SecurityGroups))
+	d.Set("subnets", flattenSubnetsFromAvailabilityZones(alb.AvailabilityZones))
+	d.Set("vpc_id", alb.VpcId)
+	d.Set("zone_id", alb.CanonicalHostedZoneId)
+	d.Set("dns_name", alb.DNSName)
+
+	respTags, err := elbconn.DescribeTags(&elbv2.DescribeTagsInput{
+		ResourceArns: []*string{alb.LoadBalancerArn},
+	})
+	if err != nil {
+		return errwrap.Wrapf("Error retrieving ALB Tags: {{err}}", err)
+	}
+
+	var et []*elbv2.Tag
+	if len(respTags.TagDescriptions) > 0 {
+		et = respTags.TagDescriptions[0].Tags
+	}
+	d.Set("tags", tagsToMapELBv2(et))
+
+	attributesResp, err := elbconn.DescribeLoadBalancerAttributes(&elbv2.DescribeLoadBalancerAttributesInput{
+		LoadBalancerArn: aws.String(d.Id()),
+	})
+	if err != nil {
+		return errwrap.Wrapf("Error retrieving ALB Attributes: {{err}}", err)
+	}
+
+	accessLogMap := map[string]interface{}{}
+	for _, attr := range attributesResp.Attributes {
```

Comment:
Wouldn't this be easier to maintain going forward if we just had a single `attributes` (`TypeMap`) field instead of trying to do this conversion?

Admittedly we'd probably still need to have some kind of key-filtering because of https://github.com/hashicorp/terraform/pull/8104